### PR TITLE
Add `ocaml-nn` to the Machine Learning libraries section

### PR DESCRIPTION
**Pull request by ck090**
Adding a new library in the machine learning space. This library implements a Fully-Connected Neural Network using purely functional programming in OCaml. Provides lots of customizing functions such as loss functions, optimizers, and gradient descent methods. 

Train and run a neural network on any dataset. We implement a fully-connected multi-layered neural network that learns network parameters using back propagation implemented as a gradient descent algorithm. The data structure we use is OCaml Arrays. We further provide the following hyper-parameter customizability:

- Optimization Functions:  `Vanilla GD | GD w/Momentum | RMS Prop` <br>
- Activation Functions:  `ReLU | TanH | Sigmoid ` <br>
- Gradient Descent Type:  `Stochastic GD | Mini-Batch GD | Vanilla GD` <br>
- Learning Rate: `<any floating point number, ideally 0.05/0.1>` <br>
- Epochs: `<any integer greater than 0, ideally 1000>` <br>
- Beta1 and Beta2: `<any floating point number in [0,1]>` <br>
- Number of Hidden Units: `<any integer greater than 0, ideally 10>` <br>
- Epsilon: `<any floating point number, ideally 1e-8>` <br>

We run some sample tests on 4000 rows of the MNIST-10 dataset and test on 20 rows. The results are as follows:

| Optimization  | Accuracy | Time (sec) |
|---------------|----------|------------|
| Vanilla GD    | 0.842    | 65         |
| GD w/Momentum | 0.894    | 55         |
| RMS Prop      | 0.910    | 60         |

### CODE REVIEW

**Comment by rizo** [comment]
Hi @ck090! Apologies for the delay.

This looks like a great library. Thank you for adding it.
